client/src/language/languageClient.ts

```diff
@@ -60,6 +60,9 @@ export async function activateLanguageServer (context: ExtensionContext): Promis
     // Register the server for bitbake documents
     // TODO: check new documentSelector
     documentSelector: [{ scheme: 'file', language: 'bitbake' }],
+    synchronize: {
+      configurationSection: 'bitbake'
```

Comment:
Fixed. Seems much less convenient, thought. Makes me wonder if I am missing something.

Reply:
The vscode sends the same notification type inherently on setting changes. Do we really need to manually do that again? And the settings from the getConfiguration() doesn't specify that it is the bitbake settings we want.

Plus, we already have the event handler registered for it: client/src/extension.ts:108

Reply:
According to my experimentations,  the settings on the server are currently never initialized or updated. Do you obtain the same thing?

I admit the `onDidChangeConfiguration` in extension.ts missed to my attention. I was about to move the new code, then I realized the parent function is called "activateLanguageServer". I'm thinking it is better to keep the communication between client and the server encapsulated into that function.

I am sending the whole configuration because the handler is called `onDidChangeConfiguration` and not `onDidChangeBitBakeConfiguration`. I've been hesitating to do so, but I've been thinking by doing otherwise someone might eventually waste time trying to understand why they're not receiving the whole configuration, and it was not worth the performance benefits that are probably imperceptible.

Reply:
The settings are updated accordingly as expected on my side.

The function name is called `activateLanguageServer` but it actually starts the client, but the client will initiate the server behind the scene because the client is created with the context of the server. And it is recommended to push all these workspace event handlers in the `context.subscriptions` since they are disposables.

The `onDidChangeConfiguration` is supposed to be listening to the changes that were actually changed. and our language server doesn't rely on non-bitbake settings.
![image](https://github.com/yoctoproject/vscode-bitbake/assets/47988425/8fa37691-1124-4f0b-98ff-4416144cd6e7)

![image](https://github.com/yoctoproject/vscode-bitbake/assets/47988425/9a5cb433-3007-4581-a5f5-c2b4efed36de)

Reply:
I had a misunderstanding, the option `synchronized` is deprecated but it is the one that makes the event handler `onDidChangeConfiguration` work on the server side (No need to send it specifically from the client, Eugen's had it at the beginning). Just I haven't figured out how to properly use the new model `workspace/configuration` as stated in the vscode-client lib

Reply:
https://github.com/microsoft/vscode-languageserver-node/blob/main/protocol/src/common/protocol.configuration.ts#L14

According to this description from the link above and my experiments, I'd say we keep using the option `synchronize` to automatically keep the configurations synced with the server. If we want to use the request model `workspace/configuration` and want the server to be notified of the config change, we'd still need to have another communication about the change (we could use the existing `onDidChangeConfiguration` or our own). This is just a redundant step in my opinion.

Reply:
It is tedious to notify the server of config change just to let the server send another request to fetch it. Though the `synchronize` is marked as deprecated we may use it as long it doesn't break our extension.